aptos-move/e2e-move-tests/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+            .payload(payload)
+            .sign();
+        let output = self.executor.execute_transaction(txn);
+        if matches!(output.status(), TransactionStatus::Keep(_)) {
+            self.executor.apply_write_set(output.write_set());
+            self.txn_seq_no.insert(*account.address(), seq_no + 1);
+        }
+        output.status().to_owned()
+    }
+
+    /// Run the specified entry point `fun`. Arguments need to be provided in bcs-serialized form.
+    pub fn run_entry_txn(
+        &mut self,
+        account: &Account,
+        fun: MemberId,
+        ty_args: Vec<TypeTag>,
+        args: Vec<Vec<u8>>,
+    ) -> TransactionStatus {
+        let MemberId {
+            module_id,
+            member_id: function_id,
+        } = fun;
+        self.run_txn(
+            account,
+            TransactionPayload::ScriptFunction(ScriptFunction::new(
+                module_id,
+                function_id,
+                ty_args,
+                args,
+            )),
+        )
+    }
+
+    /// Publishes the Move Package found at the given path on behalf of the given account.
+    pub fn publish_package(
+        &mut self,
+        account: &Account,
+        path: &Path,
+        upgrade_policy: UpgradePolicy,
+    ) -> TransactionStatus {
+        let package = BuiltPackage::build(MovePackageDir::new(path.to_owned()), true, false)
+            .expect("building package must succeed");
+        let code = package.extract_code();
+        let metadata = package
+            .extract_metadata(upgrade_policy)
+            .expect("extracting package metdata must succeed");
+        self.run_txn(
+            account,
+            aptos_stdlib::code_publish_package_txn(
+                bcs::to_bytes(&metadata).expect("PackageMetadata has BCS"),
+                code,
+            ),
+        )
+    }
+
+    /// Reads the raw, serialized data of a resource.
+    pub fn read_resource_raw(&self, addr: &AccountAddress, struct_id: MemberId) -> Option<Vec<u8>> {
+        let path = access_path::AccessPath::new(
+            *addr,
+            bcs::to_bytes(&access_path::Path::Resource(StructTag {
+                address: *struct_id.module_id.address(),
+                module: struct_id.module_id.name().to_owned(),
+                name: struct_id.member_id.to_owned(),
+                type_params: vec![],
+            }))
+            .unwrap(),
+        );
+        self.executor
+            .read_from_access_path(&path)
+            .and_then(|bytes| if bytes.is_empty() { None } else { Some(bytes) })
+    }
+
+    /// Reads the resource data `T`.
+    pub fn read_resource<T: DeserializeOwned>(
+        &self,
+        addr: &AccountAddress,
+        struct_id: MemberId,
+    ) -> Option<T> {
+        Some(
+            bcs::from_bytes::<T>(&self.read_resource_raw(addr, struct_id)?).expect(
+                "serialization expected to succeed (Rust type incompatible with Move type?)",
+            ),
+        )
+    }
+
```

Comment:
add read table resource support like here https://github.com/aptos-labs/aptos-core/blob/04fcf5107f025ce955c3c56d7c6b488fa1651a29/aptos-move/aptos-transactional-test-harness/src/aptos_test_harness.rs#L884?

Reply:
Good idea, but I'd suggest to incrementally extend this as need arises (so next PR). Right now the thing has the bare minimum to handle the existing test scenarios.